src/fsharp/FSharp.Compiler.Service.ProjectCracker/Program.fs

```diff
@@ -0,0 +1,414 @@
+              | Some "X86", _, _ -> yield "--platform:x86"
+              | Some "X64", _, _ -> yield "--platform:x64"
+              | Some "ITANIUM", _, _ -> yield "--platform:Itanium"
+              | _ -> ()
+              match targetTypeOpt |> Option.map (fun o -> o.ToUpperInvariant()) with
+              | Some "LIBRARY" -> yield "--target:library"
+              | Some "EXE" -> yield "--target:exe"
+              | Some "WINEXE" -> yield "--target:winexe"
+              | Some "MODULE" -> yield "--target:module"
+              | _ -> ()
+              yield! otherFlags
+              for f in resources do
+                  yield "--resource:" + f
+              for i in libPaths do
+                  yield "--lib:" + mkAbsolute directory i 
+              for r in references do
+                  yield "-r:" + r 
+              yield! files ]
+    
+      member x.Options = options
+      member x.FrameworkVersion = fxVer
+      member x.ProjectReferences = projectReferences
+      member x.References = references
+      member x.CompileFiles = files
+      member x.ResourceFiles = resources
+      member x.EmbeddedResourceFiles = embeddedResources
+      member x.ContentFiles = content
+      member x.OtherFiles = noaction
+      member x.PageFiles = pages
+      member x.OutputFile = outFileOpt
+      member x.Directory = directory
+      member x.AssemblyName = assemblyNameOpt
+      member x.OutputPath = outputPathOpt
+      member x.FullPath = fsprojFullPath
+      member x.LogOutput = logOutput
+      static member Parse(fsprojFileName:string, ?properties, ?enableLogging) = new FSharpProjectFileInfo(fsprojFileName, ?properties=properties, ?enableLogging=enableLogging)
+
+
+  let getOptions file enableLogging properties =
+    let rec getOptions file : Option<string> * ProjectOptions =
+      let parsedProject = FSharpProjectFileInfo.Parse(file, properties=properties, enableLogging=enableLogging)
+      let referencedProjectOptions =
+        [| for file in parsedProject.ProjectReferences do
+             if Path.GetExtension(file) = ".fsproj" then
+                match getOptions file with
+                | Some outFile, opts -> yield outFile, opts
+                | None, _ -> () |]
+
+      let options = { ProjectFile = file
+                      Options = Array.ofList parsedProject.Options
+                      ReferencedProjectOptions = referencedProjectOptions
+                      LogOutput = parsedProject.LogOutput }
+      parsedProject.OutputFile, options
+
+    snd (getOptions file)
+
+  let addMSBuildv14BackupResolution () =
+    let onResolveEvent = new ResolveEventHandler(fun sender evArgs ->
+      let requestedAssembly = AssemblyName(evArgs.Name)
+      if requestedAssembly.Name.StartsWith("Microsoft.Build") &&
+          not (requestedAssembly.Name.EndsWith(".resources")) then
+        requestedAssembly.Version <- Version("14.0.0.0")
+        Assembly.Load (requestedAssembly)
```

Comment:
No, each event handler (here only one) is called in turn, if they all return `null` then an exception is raised as normal as if there had been no event handlers in place.